crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -1818,14 +1818,9 @@ impl OlmMachine {
                 .await;
         }
 
-        let event = serde_json::from_value::<Raw<AnyTimelineEvent>>(decrypted_event.into())?;
+        let event = serde_json::from_value::<Raw<AnyMessageLikeEvent>>(decrypted_event.into())?;
```

Comment:
I know state events aren't currently encrypted, but does the change from `AnyTimelineEvent` to `AnyMessageLikeEvent` here mean that we are assuming this is not a state event? Are we OK with that?

Reply:
Good question.

Yes I am assuming this is not a state event, and that our lack of encryption for state events is not going to change any time soon. It felt like using the more specific type might be more useful for callers. It's also worth noting that situation can't change without changes elsewhere in the crypto crate (specifically we construct the JSON for the decrypted event in `InboundGroupSession::decrypt`).

On the other hand: maybe it's better we encourage callers to be ready for state events by continuing to return `Raw<AnyTimelineEvent>`.

I'd be interested in @poljar's thoughts here.

Reply:
I really don't expect state events to be able to use the same mechanisms for encryption as we use for non-state events, perhaps I'm mistaken here but I would expect that they would be different enough to warrant a separate method.

Reply:
>  I would expect that they would be different enough to warrant a separate method.

I guess the point is that it's not just about a separate method: the intention is to embed `DecryptedRoomEvent` inside `TimelineEvent` and `SyncTimelineEvent` -- so we'll be hardcoding this knowledge there too.

That said: on the grand scale of changes that will be needed to support encrypted state events, I don't think changing this later is going to be that significant.